machinery-helper/machinery_helper.go

```diff
@@ -236,11 +262,84 @@ func findUnmanagedFiles(dir string, rpmFiles map[string]string, rpmDirs map[stri
 	}
 }
 
+func amendMode(entry *UnmanagedFile, perm os.FileMode) {
+	if entry.Type == "link" {
+		return
+	}
+
+	result := int64(perm.Perm())
+
+	if perm&os.ModeSticky > 0 {
+		result |= 01000
+	}
+	if perm&os.ModeSetuid > 0 {
+		result |= 04000
+	}
+	if perm&os.ModeSetgid > 0 {
+		result |= 02000
+	}
+	entry.Mode = strconv.FormatInt(result, 8)
+}
+
+func dirInfo(path string) (size int64, file_count int) {
+	files, _ := readDir(path)
+
+	size = int64(0)
+	file_count = len(files)
+	for _, f := range files {
+		if f.IsDir() {
+			if _, ok := IgnoreList[path + f.Name()]; !ok {
+				sub_size, sub_count := dirInfo(path + f.Name() + "/")
```

Comment:
don't use underscores in Go names; var sub_size should be subSize<br>don't use underscores in Go names; var sub_count should be subCount